change-validation/src/main/scala/com/normation/plugins/changevalidation/ChangeRequestJdbcRepository.scala

```
@@ -291,29 +299,24 @@ class WoChangeRequestJdbcRepository(
   /**
    * Update a change request. The change request must exists.
    */
-  def updateChangeRequest(changeRequest: ChangeRequest, actor: EventActor, reason: Option[String]): Box[ChangeRequest] = {
+  def updateChangeRequest(changeRequest: ChangeRequest, actor: EventActor, reason: Option[String]): IOResult[ChangeRequest] = {
     // no transaction between steps, because we don't actually use anything in the existing change request
 
     for {
-      cr      <- roRepo.get(changeRequest.id)
-      ok      <- cr match {
-                   case None    =>
-                     val msg = s"Cannot update non-existent Change Request with id ${changeRequest.id.value}"
-                     ChangeValidationLogger.warn(msg)
-                     Failure(msg)
-                   case Some(x) => Full("ok")
-                 }
-      update  <- {
+      _       <- roRepo
+                   .get(changeRequest.id)
+                   .notOptional(s"Cannot update non-existent Change Request with id ${changeRequest.id.value}")
+                   .tapError(err => ChangeValidationLoggerPure.error(err.fullMsg))
+      _       <- {
```

replaced the call to `ChangeValidationLogger.warn` with `ChangeValidationLoggerPure.error` since the block returns a Failure (instead of a Full accompanied by a warning). do tell me whether i should change it back